### PR TITLE
[INLONG-5281][TubeMQ] Reconnect to tubemq server if StandbyException is thrown

### LINKED ISSUE #5281: [Bug][TubeMQ] After running for a period of time, it has been unable to connect to the broker
### What happened

I use tubemq-client-go to consume tubemq's msg.  After running for a period of time, it has been unable to connect to the broker.  The phenomenon of not being able to connect to the broker lasted for 8 hours, and it was connected after restarting the process.

go mod version is  github.com/apache/incubator-inlong/inlong-tubemq/tubemq-client-twins/tubemq-client-go v0.0.0-20220311090829-1838ba3eb272。

The key log is as follows：

2022-07-28 11:44:23.548 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xxx.xxx.xx.xx:8123: connect: connection refused
2022-07-28 11:44:24.155 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xx.xxx.xxx.xx:8123: connect: connection refused
2022-07-28 11:44:24.947 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xxx.xxx.xx.xx:8123: connect: connection refused
2022-07-28 11:44:25.556 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xx.xxx.xxx.xx:8123: connect: connection refused
2022-07-28 11:44:26.148 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xxx.xxx.xx.xx:8123: connect: connection refused
2022-07-28 11:44:26.957 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xx.xxx.xxx.xx:8123: connect: connection refused
2022-07-28 11:44:27.346 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xxx.xxx.xx.xx:8123: connect: connection refused
2022-07-28 11:44:28.355 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xx.xxx.xxx.xx:8123: connect: connection refused
11:44:28.355 INFO    client/consumer_impl.go:249     [CONSUMER]GetMessage error code: 5, msg:dial tcp xx.xxx.xxx.xx:8123: connect: connection refused
2022-07-28 11:44:32.859 ERROR   client/heartbeat.go:103 consumer hb err code: 2, msg:exceptionName:"org.apache.tubemq.corerpc.exception.StandbyException"  stackTrace:"xxxxx:8609 is not master now. the connecting client id is xxx_xxx_xxx.-25120-1657705121000-1-go-0.1.1"

### What you expected to happen

The master-slave switch can connect to the broker normally.

### How to reproduce

I dont know. 
In this case, it has been unable to connect after more than 2 months of running.

### Environment

os is  centos 7 
go mod version is  github.com/apache/incubator-inlong/inlong-tubemq/tubemq-client-twins/tubemq-client-go v0.0.0-20220311090829-1838ba3eb272。

### InLong version

master

### InLong Component

InLong TubeMQ

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TszKitLo40**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #5281 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*
Reconnect to tubemq server if StandbyException is thrown
### Modifications

*Describe the modifications you've done.*
if StandbyException is returned from the tubemq server, try to reconnect to the another master.
### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by healchow — APPROVED** [review]

**Review by charlely — APPROVED** [review]
LGTM

File: inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go [hunk 1]
```diff
@@ -129,9 +129,9 @@ func (h *heartbeatManager) consumerHB2Master() {
 				return
 			}
 		}
+		h.consumer.masterHBRetry = 0
```
Comment (charlely) [inline inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:132]: If HBRetry over max tims， need use other master to retry connect. 
But HBRetry over max cann't to change, if the master is downoff.

**Review by dockerzhang — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

File: inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go (hunk 1 above)
Comment (gosonzhang) [reply inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:132]: @TszKitLo40, how about this problem?

Comment (TszKitLo40) [reply inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:132]: In https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L566
`master_sh_retry_cnt_` in C++ SDK is the same meaning of `masterHBRetry` in Go SDK. But it seems that C++ SDK also does not try to reconnect to another address if `master_sh_retry_cnt_` exceeds the `config_.GetMaxHeartBeatRetryTimes`, it just enlarge the value. Or do I miss some logic of C++ SDK? @gosonzhang

Comment (gosonzhang) [reply inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:132]: @TszKitLo40, there should be a logical omission:
1. The master address obtained by each heartbeat is performed through getCurrentMasterAddr()[1];
2. In getCurrentMasterAddr(), according to the valid master address obtained during registration [2], locate the heartbeat through the curr_master_addr_ parameter;
3. The curr_master_addr_ value is effectively determined through the registration between the client and the Master [3]

1. https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L510
2. https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L1414
3. https://github.com/apache/inlong/blob/master/inlong-tubemq/tubemq-client-twins/tubemq-client-cpp/src/baseconsumer.cc#L374

Comment (gosonzhang) [reply inlong-tubemq/tubemq-client-twins/tubemq-client-go/client/heartbeat.go:132]: image

In go, miss this logic

**Pull request merged**